buildSrc/src/main/java/edu/wpi/gripgenerator/settings/DefinedParamType.java

```diff
@@ -66,8 +67,8 @@ public DefinedParamType(String type, Parameter param) {
     }
 
     public DefinedParamType(String type, DefinedParamState state) {
-        this.type = type;
-        this.state = state;
+        this(type);
+        this.state = Optional.ofNullable(state);
```

Comment:
Why is the state optional?  It seems like something like that would be required.  What happens to a socket if it doesn't have a state?  (Also, why is it called "state" and not "direction"?)